test/bigtable/row.js

```diff
@@ -0,0 +1,785 @@
+        a: 'a',
+        b: 'b'
+      };
+
+      row.get = function(callback) {
+        row.metadata = metadata;
+        callback(null, row, response);
+      };
+
+      row.getMetadata(function(err, metadata, apiResponse) {
+        assert.ifError(err);
+        assert.strictEqual(metadata, metadata);
+        assert.strictEqual(response, apiResponse);
+        done();
+      });
+    });
+  });
+
+  describe('increment', function() {
+    var COLUMN_NAME = 'a:b';
+    var spy;
+
+    beforeEach(function() {
+      spy = sinon.stub(Row, 'formatFamilies_').returns({
+        a: {
+          b: [{
+            value: 10
+          }]
+        }
+      });
+    });
+
+    it('should provide the proper request options', function(done) {
+      row.createRules = function(reqOpts) {
+        assert.strictEqual(reqOpts.column, COLUMN_NAME);
+        assert.strictEqual(reqOpts.increment, 1);
+        done();
+      };
+
+      row.increment(COLUMN_NAME, assert.ifError);
+    });
+
+    it('should optionally accept an increment amount', function(done) {
+      var increment = 10;
+
+      row.createRules = function(reqOpts) {
+        assert.strictEqual(reqOpts.column, COLUMN_NAME);
+        assert.strictEqual(reqOpts.increment, increment);
+        done();
+      };
+
+      row.increment(COLUMN_NAME, increment, assert.ifError);
+    });
+
+    it('should return an error to the callback', function(done) {
+      var error = new Error('err');
+      var response = {};
+
+      row.createRules = function(r, callback) {
+        callback(error, response);
+      };
+
+      row.increment(COLUMN_NAME, function(err, value, apiResponse) {
+        assert.strictEqual(err, error);
+        assert.strictEqual(value, null);
+        assert.strictEqual(apiResponse, response);
+        done();
+      });
+    });
+
+    it('should pass back the updated value to the callback', function(done) {
+      var fakeValue = 10;
+      var response = {
+        key: 'fakeKey',
+        families: [{
+          name: 'a',
+          columns: [{
+            qualifier: 'b',
+            cells: [{
+              timestampMicros: Date.now(),
+              value: fakeValue,
+              labels: []
+            }]
+          }]
+        }]
+      };
+
+      row.createRules = function(r, callback) {
+        callback(null, response);
+      };
+
+      row.increment(COLUMN_NAME, function(err, value, apiResponse) {
+        assert.ifError(err);
+        assert.strictEqual(value, fakeValue);
+        assert.strictEqual(apiResponse, response);
+        assert.strictEqual(spy.callCount, 1);
+        assert(spy.calledWithExactly(response.families));
+        done();
+      });
+    });
+  });
+
+  describe('save', function() {
+    it('should insert a key value pair', function(done) {
+      var key = 'a:b';
+      var value = 'c';
+
+      var expectedData = {
+        a: {
+          b: 'c'
+        }
+      };
+
+      row.parent.mutate = function(entry) {
```

Comment:
Part 1: Still outstanding (in that we also need to verify `parseColumnName` was called correctly)
Part 2: Done